packages/utils/src/lib/filter.ts

```diff
@@ -0,0 +1,24 @@
+export function filterItemRefsBy<T extends { refs: object[] }>(
+  items: T[],
+  refFilterFn?: (item: T['refs'][number]) => boolean,
+) {
+  return filterBy(
+    items
+      .map(item => ({
+        ...item,
+        refs: filterBy(item.refs, refFilterFn),
+      }))
+      // remove item with empty refs
+      .filter(item => item.refs.length),
+  );
+}
+
+export function filterBy<T = unknown>(
+  refs: T[] = [],
+  filterFn?: (t: T) => boolean,
+): T[] {
+  if (filterFn === undefined || refs.length === 0) {
+    return refs;
+  }
+  return refs.filter(filterFn);
+}
```

Comment:
I had a realization - this `filterBy` function doesn't need to exist :sweat_smile:  I thought the reason for it was could apply a filter optionally (`filterBy(arr, fn)` instead of `fn ? arr.filter(fn) : arr`), but actually you're never doing that in the production code. So actually you can just replace `filterBy(arr, fn)` with `arr.filter(fn)` and remove this wrapper function completely. It just introduces an unnecessary abstraction, and `Array.filter` is smarter with regards to type inference anyway (e.g. supports type predicates).

Reply:
deleted it